oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/DocumentStoreIndexerBase.java

```diff
@@ -133,6 +135,52 @@ public NodeStateEntryTraverser create(LastModifiedRange lastModifiedRange) {
         }
     }
 
+    private FlatFileStore buildFlatFileStore(NodeState checkpointedState, CompositeIndexer indexer) throws IOException {
+
+        Stopwatch flatFileStoreWatch = Stopwatch.createStarted();
+        int executionCount = 1;
+        CompositeException lastException = null;
+        List<File> previousDownloadDirs = new ArrayList<>();
+        FlatFileStore flatFileStore = null;
+        //TODO How to ensure we can safely read from secondary
+        DocumentNodeState rootDocumentState = (DocumentNodeState) checkpointedState;
+        DocumentNodeStore nodeStore = (DocumentNodeStore) indexHelper.getNodeStore();
+
+        DocumentStoreSplitter splitter = new DocumentStoreSplitter(getMongoDocumentStore());
+        List<Long> lastModifiedBreakPoints = splitter.split(Collection.NODES, 0L ,10);
+        FlatFileNodeStoreBuilder builder = null;
+
+        while (flatFileStore == null && executionCount <= MAX_DOWNLOAD_RETRIES) {
+            try {
+                //TODO Use flatFileStore only if we have relative nodes to be indexed
```

Comment:
Not quite sure why this TODO was+is there... I would remove it

Reply:
done.